edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -717,29 +679,15 @@ impl ModuleRuntime for DockerModuleRuntime {
             future::join_all(n).map(|_| ())
         }))
     }
+}
 
-    fn top(&self, id: &str) -> Self::TopFuture {
-        let id = id.to_string();
-        Box::new(
-            self.client
-                .container_api()
-                .container_top(&id, "")
-                .then(|result| match result {
-                    Ok(resp) => {
-                        let p = parse_top_response::<Deserializer>(&resp).with_context(|_| {
-                            ErrorKind::RuntimeOperation(RuntimeOperation::TopModule(id.clone()))
-                        })?;
-                        Ok(ModuleTop::new(id, p))
-                    }
-                    Err(err) => {
-                        let err = Error::from_docker_error(
-                            err,
-                            ErrorKind::RuntimeOperation(RuntimeOperation::TopModule(id)),
-                        );
-                        Err(err)
-                    }
-                }),
-        )
+impl Authenticator for DockerModuleRuntime {
+    type Error = Error;
+    type Request = Request<Body>;
+    type AuthenticateFuture = Box<dyn Future<Item = AuthId, Error = Self::Error> + Send>;
+
+    fn authenticate(&self, req: &Self::Request) -> Self::AuthenticateFuture {
+        authenticate(self, req)
```

Comment:
Yes, it is. The function `authenticate` is used in 2 places: here and for TestModuleList implementation. This mocking "pattern" used for `list_with_details` method tests prior to my changes. I'm not sure whether it is a good approach though.

Reply:
Yeah, the alternative is to make it a free `pub fn` and expect users to reference it directly. I did it this way for `list_with_details` because it's more easily discoverable when it's in the trait's namespace than a standalone `pub fn`.